### DISCUSSION TITLE
MeshCentral Does Not Send Notifications to NTFY with Basic Authorization

### DISCUSSION THREAD

**Discussion by Creogenesis** [Ylianst/MeshCentral#5571] [Issues] [answered]
**MeshCentral** and **NTFY** are deployed on a VPS as separate **Docker Swarm stack containers**. Individually, these services operate without errors and have been time-tested.

Here is the configuration for NTFY within MeshCentral (found on a YouTube tutorial by the author and in the documentation):

```
"messaging": {
  "ntfy": true,
  "ntfy": {
     "host": "https://myserverntfy.com",
     "authorization": "Basic ----------------"
  }
},
"_smtp": {
```

This block is inserted after the domains section. The Basic token was obtained using the command **echo -n 'username:passw@rd' | base64**
Additionally, for testing purposes, a notification was successfully sent from the MeshCentral container (**docker exec -it namecontainer sh**) using the command:

```
wget --post-data="title=Test&message=This is a test message from MeshCentral container" \
     --header="Authorization: Basic ------------------------" \
     https://myserverntfy.com/topicformeshcentral
```

This method of sending works!
Sending from the standard VPS console also works in two ways - via **token** and **Basic auth**.

I'm seeking assistance. It could be an error in my configuration, or it might be a bug. I would like to resolve this issue, as I've been struggling with it all day :)

Also, I could not find a clear explanation of what the **userurl** option means in the ntfy section.

**Comment by si458** [comment, marked as answer]
your config.json is incorrect, the should only be 1 ntfy option

so it should be 
```
"messaging": {
  "ntfy": {
     "host": "https://myserverntfy.com",
     "authorization": "Basic ----------------"
  }
},
"_smtp": {
```

you only set `"ntfy": true` if you want to use the normal public ntfy server not your own!

edit: also the youtube video was made BEFORE i added the `authorization` section the other month!
also let me know which docs u followed and ill check if i may have made a BOOBOO with explaining it

EDIT2: also the host should be JUST THE DOMAIN NAME and NOT INCLUDE https:// 
`const url = 'https://' + (((typeof parent.config.messaging.ntfy == 'object') && (typeof parent.config.messaging.ntfy.host == 'string')) ? parent.config.messaging.ntfy.host : 'ntfy.sh') + '/' + encodeURIComponent(to.substring(5));`

EDIT3: descriptions are inside the schema file or the config.md file - https://ylianst.github.io/MeshCentral/meshcentral/config/
![image](https://github.com/Ylianst/MeshCentral/assets/765314/57872443-6154-4cca-9bfc-1821b30361c2)

**Reply by Creogenesis** [reply]
"messaging": {
  "ntfy": {
     "host": "**myserverntfy.com**",
     "authorization": "Basic ----------------"
  }
},
"_smtp": {

right?

**Reply by si458** [reply]
correct and that should work after a restart of meshcentral!

**Reply by Creogenesis** [reply]
I just restarted the stack after editing the config.json file according to your recommendation. **It worked :)** 

You provide the best support I've encountered on GitHub. Your suggestion to specify the domain without **https://** did the trick. Previously, my configuration did not have "ntfy": true, and that led nowhere.

I think it would be beneficial for others in the future to clearly state how the domain should be specified in the configuration. This way, others won't have to go through the same troubles I did.

**Reply by si458** [reply]
i mean the schema does say 'host name of the ntfy server' but the new docs do say use https:// which is wrong!
https://ylianst.github.io/MeshCentral/messaging/#ntfy-setup
thanks for pointing it out, will get it changed!
P.S. dont forget to sponsor me https://github.com/sponsors/si458 ❤️

**Reply by Creogenesis** [reply]
In my case for sponsor. Message from GitHub

It appears your account may be based in a U.S.-sanctioned region. As a result, we are unable to provide private repository services and paid services for your account. GitHub has preserved, however, your access to [certain free services for public repositories](https://docs.github.com/en/github/site-policy/github-and-trade-controls#what-is-available-and-not-available). If your account has been flagged in error, and you are not located in or resident in a sanctioned region, please [file an appeal](https://docs.github.com/en/github/site-policy/github-and-trade-controls#how-is-github-ensuring-that-folks-not-living-in-andor-having-professional-links-to-the-sanctioned-countries-and-territories-still-have-access-or-ability-to-appeal). Please read about [GitHub and Trade Controls](https://docs.github.com/en/github/site-policy/github-and-trade-controls) for more information.

Please let me know another way to express my gratitude over time.
**From Russia with** ❤️

**Reply by si458** [reply]
updated docs for you!  https://ylianst.github.io/MeshCentral/messaging/#ntfy-setup

**Reply by si458** [reply]
check my github page https://github.com/si458 i have a paypal link there, if you cant donate, please dont worry! glad i could help!

**Comment by iot7755** [comment]
Hello, as per above example, I had to change the `authorization` key value to `Bearer _tk........` instead of `Basic _tk.....` and it worked for me.
